doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,281 @@
+    * Existing APIs: get_name, get_type, get_my_slot_idx, etc
+    * New APIs
+        * get_dpu_state,  (will take dpu ID or slot/offset as an argument)
+        * get_dpu_health, (will take dpu ID or slot/offset as an argument)
+        * get_dpu_id, (will slot/offset as an argument)
+        * get_dpu_ip, (will take dpu ID or slot/offset as an argument)
+        * get_dpu_mac (will take dpu ID or slot/offset as an argument)
+    * Update sensor, peripheral device date to sysfs, state-DB/ChassisStateDB (platform-DB)
+
+### 3.1. Thermal management
+* Platform  initializes all sensors
+* Thermalctld fetch CPU temperature, DPU temperature, fan speed, monitor and update the DB
+* Thermal manager reads all thermal sensor data, run thermal policy and take policy action Ex. Set fan speed, set alarm, set syslog, set LEDs 
+* Platform collects fan related data such as presence, failure and then applies fan algorithm to set the new fan speed
+* The north bound CLI/Utils/App use DB data to ”show environment”, ”show platform temp” show platform fan”
+
+Thermal management sequence diagram
+<p align="center"><img src="./images/thermal-mgmt-seq.svg"></p>
+
+### 3.2. Platform device data collection 
+* thermalcontrold, led and PSUd post device data to DB periodically
+* during the boot up of the daemons, it will collect the constant data like serial number, manufacture name, etc.
+* For the variable ones (temperature, voltage, fan speed ....) need to be collected periodically. 
+
+### 3.3. DPU STATE
+* DPU state represents the various operational states of the DPU. They are: Powered, PCIe-Link, Host-DPU Eth-Link, Firmware, OS, CP, DP.
+* The table below shows the various DPU states, the producer and consumers of them.
+#### 3.3.1. DPU_STATE - Before DPU OS boots
+* All DPU States are updated by the Host PMON
+* These states are available to host PMON as register bits from DPUs.
+* Additionally, the FPGA provides temp sensor values too
+#### 3.3.2. DPU_STATE - After DPU OS boots
+* DPU updates the DB with additional information such as the reason for every state, and timestamp 
+* Carries Die Temp and threshold
+* Has reason for previous reboot
+* Carries DP, CP up down reasons
+
+DPU_STATE Table
+
+| DPU_STATE | Powered | PCIe_Link | NPU_DPU_Ge_Link | Firmware | OS_boot |  | CtrlPlane | DataPlane |
+| --- | --- | --- | --- | --- | --- | --- | --- | --- |
+| DPU0 | on | up | up | up | up |  | up | down |
+| DPU1 | off | down | down | up | up |  | down | up |
+| Genrated By | dpu | dpu | dpu | dpu | dpu |  | dpu | dpu |
+| Updated By | ss-pmon | ss-pmon | ss-pmon | ss-pmon  | ss-pmon |  | ss-pmon | ss-pmon |
+
+Potential consumer: Switch CLIs, Utils (install/repair images), HA, LB, Life Cycle Manager 
+
+Use cases: Debuggability, error recovery (reset, power cycle) and fault management, consolidated view of Switch and DPU state/health
+
+#### 3.3.3. ChassisStateDB Schema for DPU_STATE
+```
+Table: “DPU_STATE”
+
+SCHEMA
+key:  dpu_state:1
+ 
+HMSET dpu_state:1
```

Comment:
last time when we talked, we had decided to to keep these details as vendor specific, and this can be part of health details

Reply:
Outdated